xptifw/samples/sycl_perf_collector/sycl_perf_collector.cpp

```diff
@@ -0,0 +1,891 @@
+    auto StreamID = xptiRegisterStream(stream_name);
+    xptiRegisterCallback(StreamID,
+                         (uint16_t)xpti::trace_point_type_t::wait_begin,
+                         traceCallback);
+    xptiRegisterCallback(StreamID, (uint16_t)xpti::trace_point_type_t::wait_end,
+                         traceCallback);
+    xptiRegisterCallback(StreamID,
+                         (uint16_t)xpti::trace_point_type_t::task_begin,
+                         traceCallback);
+    xptiRegisterCallback(StreamID, (uint16_t)xpti::trace_point_type_t::task_end,
+                         traceCallback);
+    xptiRegisterCallback(StreamID,
+                         (uint16_t)xpti::trace_point_type_t::barrier_begin,
+                         traceCallback);
+    xptiRegisterCallback(StreamID,
+                         (uint16_t)xpti::trace_point_type_t::barrier_end,
+                         traceCallback);
+    xptiRegisterCallback(StreamID,
+                         (uint16_t)xpti::trace_point_type_t::graph_create,
+                         graphCallback);
+    xptiRegisterCallback(StreamID,
+                         (uint16_t)xpti::trace_point_type_t::node_create,
+                         graphCallback);
+    xptiRegisterCallback(StreamID,
+                         (uint16_t)xpti::trace_point_type_t::edge_create,
+                         graphCallback);
+  } else if (std::string("sycl.experimental.mem_alloc") == stream_name &&
+             Check) {
+    auto StreamID = xptiRegisterStream(stream_name);
+    xptiRegisterCallback(StreamID,
+                         (uint16_t)xpti::trace_point_type_t::node_create,
+                         graphMemCallback);
+    xptiRegisterCallback(StreamID,
+                         (uint16_t)xpti::trace_point_type_t::edge_create,
+                         graphMemCallback);
+    xptiRegisterCallback(StreamID,
+                         (uint16_t)xpti::trace_point_type_t::mem_alloc_begin,
+                         syclMemCallback);
+    xptiRegisterCallback(StreamID,
+                         (uint16_t)xpti::trace_point_type_t::mem_alloc_end,
+                         syclMemCallback);
+    xptiRegisterCallback(StreamID,
+                         (uint16_t)xpti::trace_point_type_t::mem_release_begin,
+                         syclMemCallback);
+    xptiRegisterCallback(StreamID,
+                         (uint16_t)xpti::trace_point_type_t::mem_release_end,
+                         syclMemCallback);
+  } else if (std::string("sycl.pi") == stream_name && Check) {
+    auto StreamID = xptiRegisterStream(stream_name);
+    xptiRegisterCallback(StreamID,
+                         (uint16_t)xpti::trace_point_type_t::function_begin,
+                         syclPiCallback);
+    xptiRegisterCallback(StreamID,
+                         (uint16_t)xpti::trace_point_type_t::function_end,
+                         syclPiCallback);
+  } else if (std::string("sycl.perf") == stream_name && Check) {
```

Comment:
what is sycl.perf and sycl.perf.detail is enabled at the same time? will we handle them proper;y in the same callback function?